tck/src/main/java/org/eclipse/microprofile/openapi/tck/utils/TCKMatchers.java

```diff
@@ -169,4 +173,21 @@ private static BigDecimal toBigDecimal(Number number) {
         }
 
     }
+
+    /**
+     * Creates a matcher which matches a map entry with an allowed value or the absence of a map entry from the object.
+     *
+     * @param entryName
+     *            name of the entry in the map
+     * @param validValue
+     *            a single allowed value in the map entry, if present
+     * @return the matcher
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> Matcher<T> hasOptionalEntry(String entryName, Object validValue) {
+        Matcher<T> hasEntry = (Matcher<T>) hasEntry(entryName, validValue);
+        Matcher<T> entryMissing = (Matcher<T>) not(hasKey(entryName));
```

Comment:
Hmm, I think this ends up matching anything that isn't a map, as well as a map without the given key, which probably isn't what we want given how difficult it sometimes is to tell what restassured is going to give us to match against.

Reply:
I'll look into that further.

Reply:
Maybe we need something like `both(isA(Map.class)).and(either(hasEntry).or(entryMissing));`

Reply:
This is complete. Using `both` did not work so I went with `allOf` instead.